internal/experimental/action/registry_login.go

```diff
@@ -18,21 +18,27 @@ package action
 
 import (
 	"io"
+
+	"helm.sh/helm/v3/internal/experimental/registry"
+	"helm.sh/helm/v3/pkg/action"
 )
 
 // RegistryLogin performs a registry login operation.
 type RegistryLogin struct {
-	cfg *Configuration
+	cfg *action.Configuration
```

Comment:
This whole file is moved out of action and into internal/experimental. I'm just putting this here because we not call into the `action` package.

The functions like `NewRegistryLogin` have been in the action package and in use by others through the SDK. Some examples include [this](https://github.com/clcchai/codefresh-onprem-kcfi) and [this](https://github.com/fybrik/fybrik) (there are more). By moving the code from action to internal we are breaking the usage and not providing a functional path forward.

Reply:
Resolved in 77750962d5c803857e19032bcb5e1bec0885ba25

Reply:
FYI, this was covered in https://github.com/helm/helm/pull/9782#issuecomment-858787691, https://github.com/helm/helm/pull/9782#issuecomment-858908117 and https://github.com/helm/helm/pull/9782#issuecomment-861903870 where it was agreed that the file should never have ended up in `pkg/` and because there are experimental they should be moved there.

Reply:
Seriously, we discussed this AT LENGTH. We're trying to get as much experimental code out of `action` as we can. But whatever... just merge this thing and get it done.

Reply:
@technosophos let's have a quick chat about this.